src/aten/TensorShape.cpp

```diff
@@ -44,4 +52,183 @@ Tensor XPUNativeFunctions::unfold(
   return at::native::unfold(self, dimension, size, step);
 }
 
+inline c10::MemoryFormat cat_compute_output_memory_format(
```

Comment:
should we reuse these utilities from PyTorch, instead of copy out?

Reply:
The code is inside the cu file

Reply:
@xytintel , I think we can submit another PR to the stock PyTorch by exposing the utility function. But it should not block this PR.

Reply:
Will submit a PR to stock PyTorch.